### PR TITLE
feat: eloqua new destination cdkv2

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> eloqua new destination cdkv2

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      $.assert(messageType, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='identify' || .message.type.toLowerCase() ==='track', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.traits || .message.properties, "Message traits/properties not present. Aborting message.");
+
+  - name: prepareIdentifiersResponse
+    description: |
+      Populate traits
+    template: |
+      const identifierKey = .message.context.externalId[0].identifierType
+      const objectId = .message.context.externalId[0].type.split('-')[1].toString()
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/eloqua/procWorkflow.yaml:26]: Did we handle the condition where externalId is not available or not an array ?

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      let messageType = .message.type;
+      $.assert(messageType, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='identify' || .message.type.toLowerCase() ==='track', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.traits || .message.properties, "Message traits/properties not present. Aborting message.");
+
+  - name: prepareCommonResponse
+    description: |
+      Populate traits
+    template: |
+      const identifierKey = .message.context.externalId[0].identifierType
```
Comment (koladilip) [inline src/cdk/v2/destinations/eloqua/procWorkflow.yaml:28]: we should validate this

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    path: ../../../../v0/util
+  - path: ./utils
+
+steps:
+  - name: messageType
+    template: |
+      .message.type.toLowerCase()
+  - name: validateInput
+    template: |
+      let messageType = .message.type;
```
Comment (koladilip) [inline src/cdk/v2/destinations/eloqua/procWorkflow.yaml:19]: let messageType = .message.type.toLowerCase();
// this won't cause any error if message.type is not present

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  - name: prepareTrackResponse
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    description: |
+      Populate traits
+    template: |
+      {
+        identifierFieldName : .message.context.externalId[0].identifierType,
+        data: $.outputs.prepareCommonResponse,
+        mapDataCardsSourceField: .message.context.externalId[0].identifierType,
+        customObjectId: .message.context.externalId[0].type.split('-')[1].toString()
```
Comment (koladilip) [inline src/cdk/v2/destinations/eloqua/procWorkflow.yaml:51]: write test case for type doesn't have '-'

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      }
+  - name: buildResponseForProcessTransformation
+    description: build response
+    template: |
+      const response = $.defaultRequestConfig();
+      response.body.JSON = $.outputs.prepareIdentifyResponse@payload || $.outputs.prepareTrackResponse@payload;
```
Comment (koladilip) [inline src/cdk/v2/destinations/eloqua/procWorkflow.yaml:57]: what is this "@payload"

Comment (aashishmalik) [reply src/cdk/v2/destinations/eloqua/procWorkflow.yaml]: removed

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml (hunk 2 above)
Comment (aashishmalik) [reply src/cdk/v2/destinations/eloqua/procWorkflow.yaml:25]: this is coming from rudder-sources part fo contract, will always be present

File: src/cdk/v2/destinations/eloqua/procWorkflow.yaml (hunk 3 above)
Comment (aashishmalik) [reply src/cdk/v2/destinations/eloqua/procWorkflow.yaml:16]: it will always be present

**Review by ItsSudip — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**